assets/js/modules/adsense/components/setup/v2/SetupMain.js

```diff
@@ -0,0 +1,319 @@
+	ACCOUNT_STATUS_NONE,
+	ACCOUNT_STATUS_MULTIPLE,
+	ACCOUNT_STATUS_NO_CLIENT,
+	determineAccountID,
+	determineAccountStatus,
+	determineClientID,
+	determineSiteStatus,
+} from '../../../util/status';
+const { useSelect, useDispatch } = Data;
+
+export default function SetupMain() {
+	const viewContext = useContext( ViewContextContext );
+	const eventCategory = `${ viewContext }_adsense`;
+
+	const {
+		clearError,
+		resetAccounts,
+		resetAlerts,
+		resetClients,
+		resetSites,
+		resetURLChannels,
+		setAccountID,
+		setAccountStatus,
+		submitChanges,
+	} = useDispatch( MODULES_ADSENSE );
+
+	const [
+		isAwaitingBackgroundSubmit,
+		setIsAwaitingBackgroundSubmit,
+	] = useState( false );
+	// Submit changes for determined parameters in the background when they are valid.
+	const [ isSubmittingInBackground, setIsSubmittingInBackground ] = useState(
+		false
+	);
+
+	const siteURL = useSelect( ( select ) =>
+		select( CORE_SITE ).getReferenceSiteURL()
+	);
+	const isAdBlockerActive = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).isAdBlockerActive()
+	);
+	const accounts = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAccounts()
+	);
+	const previousAccountID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAccountID()
+	);
+	const previousClientID = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClientID()
+	);
+	const accountsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getError( 'getAccounts', [] )
+	);
+	// Check whether settings differ from server and are valid.
+	const canSubmitChanges = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).canSubmitChanges()
+	);
+
+	const accountID = determineAccountID( {
+		accounts,
+		previousAccountID,
+	} );
+
+	const clients = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getClients( accountID )
+	);
+	const alerts = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getAlerts( accountID )
+	);
+	const alertsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getError( 'getAlerts', [ accountID ] )
+	);
+
+	const clientID = determineClientID( {
+		clients,
+		previousClientID,
+	} );
+	const urlChannels = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getURLChannels( accountID, clientID )
+	);
+	const urlChannelsError = useSelect( ( select ) =>
+		select( MODULES_ADSENSE ).getErrorForSelector( 'getURLChannels', [
+			accountID,
+			clientID,
+		] )
+	);
+
+	const accountStatus = determineAccountStatus( {
+		accounts,
+		clients,
+		alerts,
+		urlChannels,
+		accountsError,
+		alertsError,
+		urlChannelsError,
+		previousAccountID,
+		previousClientID,
+	} );
+	const siteStatus = determineSiteStatus( {
+		urlChannels,
+		siteURL,
+	} );
+
+	useUpdateEffect( () => {
+		if (
+			accountID !== undefined ||
+			clientID !== undefined ||
+			accountStatus !== undefined ||
+			siteStatus !== undefined
+		) {
+			setIsAwaitingBackgroundSubmit( true );
+		}
+	}, [ accountID, clientID, accountStatus, siteStatus ] );
```

Comment:
From AC:

_The isAwaitingBackgroundSubmit flag **should be set** anytime that either the accountID, clientID, accountStatus, or siteStatus **setting changes compared to its currently saved value** from the server (and only if the new value is not undefined)._

So, it seems like it's not enough to just check whether these settings are not undefined, right? Perhaps we need to add a new selector to the `createSettingsStore` factory to allow us to check whether a specific setting has changed or not. Something similar to the `haveSettingsChanged` selector, but for an individual setting.

https://github.com/google/site-kit-wp/blob/cc845eadfc0c937959c1707b2333ce3dafea8db9/assets/js/googlesitekit/data/create-settings-store.js#L282-L286

Reply:
sounds good @eugene-manuilov . I've added the `hasSettingChanged` selector. Let me know if that works.

Reply:
Very nice! Thanks, @asvinb.